.github/workflows/Java.yml

```diff
@@ -40,6 +40,8 @@ jobs:
         uses: hendrikmuhs/ccache-action@main
         with:
           key: ${{ github.job }}
+          save: ${{ github.ref == 'refs/heads/master' || github.repository != 'duckdb/duckdb' }}
```

Comment:
I wonder if we can get rid of some hard-coded names. Detecting if a PR is a fork, or if the branch is the main branch, seems impossible from contexts only. We could at least check `github.repository_owner != 'duckdb'` , but again, this doesn't seem important enough to warrant a change here.